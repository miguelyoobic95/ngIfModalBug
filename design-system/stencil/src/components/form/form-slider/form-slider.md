---
name: Slider
category: Form Components
---

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`initialValue`|number|   |initialValue passed to the slider|
|`minimum`|number|`0`|minimum number of the slider|
|`maximum`|number|`100`|maximum number of the slider|
|`disabled`|boolean|`false`|set true to disable the slider|
|`hideLabel`|boolean|`false`|set true to hide the label|
|`hideReferences`|boolean|`false`|set true to hide the references|
|`doubleSlider`|boolean|`false`|set true to use double slider|

### InitialValue

```yoo-form-slider-initial.html
<div style="padding:10px;">
    <yoo-form-slider class="accent" initial-value="34"></yoo-form-slider>
</div>
```

### Minimum & Maximum

```yoo-form-slider-max-min.html
<div style="padding:10px;">
    <yoo-form-slider minimum="-34" class="accent" initial-value="34" maximum="134"></yoo-form-slider>
</div>
```

### Disabled

```yoo-form-slider-disabled.html
<div style="padding:10px;">
    <yoo-form-slider disabled="true" class="dark" initial-value="34" ></yoo-form-slider>
</div>
```

### Hide Label

```yoo-form-slider-hide-label.html
<div style="padding:10px;">
    <yoo-form-slider hide-label="true" class="accent" initial-value="34" ></yoo-form-slider>
</div>
```

### Hide References

```yoo-form-slider-hide-ref.html
<div style="padding:10px;">
    <yoo-form-slider hide-references="true" class="accent" initial-value="34"></yoo-form-slider>
</div>
```

### Double Slider

```yoo-form-slider-double.html
<div style="padding:10px;">
    <yoo-form-slider class="accent" initial-value="34" double-slider="true"></yoo-form-slider>
</div>
```

## CSS

|Type|Name|Description|
|---|---|---|
|Inline|`inline`|yoo-form-slider is displayed inline|
|Accent|`accent`|yoo-form-slider accent|
|Dark|`dark`|yoo-form-slider dark|
|Success|`success`|yoo-form-slider success|
|Danger|`danger`|yoo-form-slider danger|
|Info|`info`|yoo-form-slider info|
|Warning|`warning`|yoo-form-slider warning|
|Gradient Accent|`gradient-accent`|yoo-form-slider gradient-accent|
|Gradient Dark|`gradient-dark`|yoo-form-slider gradient-dark|
|Gradient Success|`gradient-success`|yoo-form-slider gradient-success|
|Gradient Danger|`gradient-danger`|yoo-form-slider gradient-danger|
|Gradient Info|`gradient-info`|yoo-form-slider gradient-info|
|Gradient Warning|`gradient-warning`|yoo-form-slider gradient-warning|

```yoo-form-slider-style.html
<div style="padding:10px;">
    <yoo-form-slider class="accent" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="dark" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="success" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="danger" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="info" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="warning" initial-value="34"></yoo-form-slider>
    <br/>
    <br/>
    <yoo-form-slider class="gradient-accent" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-dark" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-success" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-danger" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-info" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-warning" initial-value="34"></yoo-form-slider>
</div>
```

### Inline Slider

```yoo-form-slider-inline.html
<div style="padding:10px;">
    <yoo-form-slider class="accent inline" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="dark inline" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="danger inline" initial-value="34"></yoo-form-slider>
    <br/>
    <yoo-form-slider class="success inline" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="warning inline" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="info inline" initial-value="34"></yoo-form-slider>
    <br/>
    <yoo-form-slider class="gradient-accent inline" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-dark inline" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-danger inline" initial-value="34"></yoo-form-slider>
    <br/>
    <yoo-form-slider class="gradient-success inline" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-warning inline" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-info inline" initial-value="34"></yoo-form-slider>
    <br/>
    <yoo-form-slider class="gradient-success inline" double-slider="true" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-warning inline" double-slider="true" initial-value="34"></yoo-form-slider>
    <yoo-form-slider class="gradient-info inline" double-slider="true" initial-value="34"></yoo-form-slider>
</div>
```

## Events

|Attr|Description|
|---|---|---|
|`singleSliderChanged`|send new value (number) when users use move the slider|
|`doubleSliderChanged`|send new value (number) when users use move the second slider|
