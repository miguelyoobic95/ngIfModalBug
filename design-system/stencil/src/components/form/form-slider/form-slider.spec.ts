import { TestWindow } from '@stencil/core/testing';
import { YooFormSliderComponent } from './form-slider';

describe('YooFormSliderComponent', () => {
  let window;
  beforeEach(() => {
    window = new TestWindow();
  });
  it('should build', () => {
      expect(new YooFormSliderComponent()).toBeTruthy();
  });

  describe('Rendering', () => {
      it ('Should have initial value displayed', async () => {
          let element = await window.load({
              components: [YooFormSliderComponent],
              html: '<yoo-form-slider class="accent" initial-value="34"></yoo-form-slider>'
            });
          expect(element).toMatchSnapshot();
      });

      it ('Should have label hidden', async () => {
          let element = await window.load({
              components: [YooFormSliderComponent],
              html: '<yoo-form-slider hide-label="true" class="accent" initial-value="34"></yoo-form-slider>'
            });
          expect(element).toMatchSnapshot();
      });

      it ('Should have references hidden', async () => {
          let element = await window.load({
              components: [YooFormSliderComponent],
              html: '<yoo-form-slider hide-references="true" class="accent" initial-value="34"></yoo-form-slider>'
            });
          expect(element).toMatchSnapshot();
      });

      it ('Should have a double slider', async () => {
          let element = await window.load({
              components: [YooFormSliderComponent],
              html: '<yoo-form-slider double-slider="true" class="accent" initial-value="34"></yoo-form-slider>'
            });
          expect(element).toMatchSnapshot();
      });

      it ('Should have a double slider inline', async () => {
          let element = await window.load({
              components: [YooFormSliderComponent],
              html: '<yoo-form-slider double-slider="true" class="inline accent" initial-value="34"></yoo-form-slider>'
            });
          expect(element).toMatchSnapshot();
      });

      describe('State Validation', async() => {
          it('Should change value only if initial Value between bounds', async () => {
              let element = await window.load({
                  components: [YooFormSliderComponent],
                  html: '<yoo-form-slider initial-value="34"></yoo-form-slider>'
                  });
              expect(element.initialValue).toEqual(34);
              expect(element.maximum).toEqual(100);
              expect(element.minimum).toEqual(0);
              element.initialValue = 125;
              await window.flush();
              expect(element).toMatchSnapshot();
              element.value = -125;
              await window.flush();
              expect(element).toMatchSnapshot();
          });
      });
  });
  describe('Re-Rendering', () => {
      it('Should re-render when initial value is updated', async () => {
        let element = await window.load({
            components: [YooFormSliderComponent],
            html: '<yoo-form-slider initial-value="34"></yoo-form-slider>'
            });
        expect(element.initialValue).toEqual(34);
        element.initialValue = 98;
        await window.flush();
        expect(element.initialValue).toEqual(98);
        expect(element).toBeTruthy();
        expect(element).toMatchSnapshot();
      });

      it('Should re-render when maximum is updated', async () => {
        let element = await window.load({
            components: [YooFormSliderComponent],
            html: '<yoo-form-slider initial-value="34"></yoo-form-slider>'
            });
        expect(element.maximum).toEqual(100);
        element.maximum = 357;
        await window.flush();
        expect(element.maximum).toEqual(357);
        expect(element).toBeTruthy();
        expect(element).toMatchSnapshot();
      });

      it('Should re-render when minimum is updated', async () => {
          let element = await window.load({
              components: [YooFormSliderComponent],
              html: '<yoo-form-slider initial-value="34"></yoo-form-slider>'
              });
          expect(element.minimum).toEqual(0);
          element.minimum = -357;
          await window.flush();
          expect(element.minimum).toEqual(-357);
          expect(element).toBeTruthy();
        expect(element).toMatchSnapshot();
      });

      it('Should re-render when disabled is updated', async () => {
        let element = await window.load({
            components: [YooFormSliderComponent],
            html: '<yoo-form-slider initial-value="34"></yoo-form-slider>'
            });
        expect(element.disabled).toEqual(false);
        element.disabled = true;
        await window.flush();
        expect(element.disabled).toEqual(true);
        expect(element).toBeTruthy();
        expect(element).toMatchSnapshot();
      });

      it('Should re-render when hideLabel is updated', async () => {
        let element = await window.load({
            components: [YooFormSliderComponent],
            html: '<yoo-form-slider initial-value="34"></yoo-form-slider>'
            });
        expect(element.hideLabel).toEqual(false);
        element.hideLabel = true;
        await window.flush();
        expect(element.hideLabel).toEqual(true);
        expect(element).toBeTruthy();
        expect(element).toMatchSnapshot();
      });

      it('Should re-render when hideReferences is updated', async () => {
        let element = await window.load({
            components: [YooFormSliderComponent],
            html: '<yoo-form-slider initial-value="34"></yoo-form-slider>'
            });
        expect(element.hideReferences).toEqual(false);
        element.hideReferences = true;
        await window.flush();
        expect(element.hideReferences).toEqual(true);
        expect(element).toBeTruthy();
        expect(element).toMatchSnapshot();
      });

      it('Should re-render when doubleSlider is updated', async () => {
        let element = await window.load({
            components: [YooFormSliderComponent],
            html: '<yoo-form-slider initial-value="34"></yoo-form-slider>'
            });
        expect(element.doubleSlider).toEqual(false);
        element.doubleSlider = true;
        await window.flush();
        expect(element.doubleSlider).toEqual(true);
        expect(element).toBeTruthy();
        expect(element).toMatchSnapshot();
      });
  });
  describe('events', () => {
    let element;
    beforeEach( async () => {
      element = new YooFormSliderComponent();
      element.value = 1337;
    });
    it ('should emit singleSliderChanged', async () => {
      element.singleSliderChanged = {
        emit: () => {}
      };
      const changedSpy = jest.spyOn(element.singleSliderChanged, 'emit');
      element.handleEvent();
      expect(changedSpy).toHaveBeenCalledWith(1337);
    });
    it ('should emit doubleSliderChanged', async () => {
      element.doubleSliderChanged = {
        emit: () => {}
      };
      const changedSpy = jest.spyOn(element.doubleSliderChanged, 'emit');
      element.doubleSlider = true;
      element.secondValue = 42;
      element.handleEvent();
      expect(changedSpy).toHaveBeenCalledWith({lowValue: 42, highValue: 1337});
    });
  });
});
