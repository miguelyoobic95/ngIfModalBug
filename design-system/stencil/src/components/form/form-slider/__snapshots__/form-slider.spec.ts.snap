// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooFormSliderComponent Re-Rendering Should re-render when disabled is updated 1`] = `
<yoo-form-slider
  class="hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container disabled"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            disabled=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            disabled=""
            type="range"
          />
          <yoo-progress-bar
            class="hydrated"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="NaN"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Re-Rendering Should re-render when doubleSlider is updated 1`] = `
<yoo-form-slider
  class="hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          <input
            class="second-slider swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          <yoo-progress-bar
            class="hydrated"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="NaN"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Re-Rendering Should re-render when hideLabel is updated 1`] = `
<yoo-form-slider
  class="hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value label-hidden"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          
          <yoo-progress-bar
            class="hydrated"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="NaN"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Re-Rendering Should re-render when hideReferences is updated 1`] = `
<yoo-form-slider
  class="hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          
          <yoo-progress-bar
            class="hydrated"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="NaN"
          />
        </div>
        
      </div>
      
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Re-Rendering Should re-render when initial value is updated 1`] = `
<yoo-form-slider
  class="hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          
          <yoo-progress-bar
            class="hydrated"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="NaN"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Re-Rendering Should re-render when maximum is updated 1`] = `
<yoo-form-slider
  class="hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        0
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            max="357"
            min="0"
            type="range"
          />
          
          <yoo-progress-bar
            class="hydrated"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="9.523809523809524"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        357
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Re-Rendering Should re-render when minimum is updated 1`] = `
<yoo-form-slider
  class="hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        -357
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            max="100"
            min="-357"
            type="range"
          />
          
          <yoo-progress-bar
            class="hydrated"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="85.55798687089715"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        100
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Rendering Should have a double slider 1`] = `
<yoo-form-slider
  class="accent hydrated"
  data-yoo-form-slider-host=""
  double-slider="true"
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          <input
            class="second-slider swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          <yoo-progress-bar
            class="accent"
            data-yoo-form-slider=""
            hide-progress="true"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Rendering Should have a double slider inline 1`] = `
<yoo-form-slider
  class="inline accent hydrated"
  data-yoo-form-slider-host=""
  double-slider="true"
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
        style="flex-direction: row"
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
          
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          <input
            class="second-slider swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          <yoo-progress-bar
            class="inline accent"
            data-yoo-form-slider=""
            hide-progress="true"
          />
        </div>
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
      </div>
      <label
        data-yoo-form-slider=""
      >
        
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Rendering Should have initial value displayed 1`] = `
<yoo-form-slider
  class="accent hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          
          <yoo-progress-bar
            class="accent"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="NaN"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Rendering Should have label hidden 1`] = `
<yoo-form-slider
  class="accent hydrated"
  data-yoo-form-slider-host=""
  hide-label="true"
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value label-hidden"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          
          <yoo-progress-bar
            class="accent"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="NaN"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Rendering Should have references hidden 1`] = `
<yoo-form-slider
  class="accent hydrated"
  data-yoo-form-slider-host=""
  hide-references="true"
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            type="range"
          />
          
          <yoo-progress-bar
            class="accent"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="NaN"
          />
        </div>
        
      </div>
      
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Rendering State Validation Should change value only if initial Value between bounds 1`] = `
<yoo-form-slider
  class="hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        0
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            max="100"
            min="0"
            type="range"
          />
          
          <yoo-progress-bar
            class="hydrated"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="0"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        100
      </label>
    </div>
  </div>
</yoo-form-slider>
`;

exports[`YooFormSliderComponent Rendering State Validation Should change value only if initial Value between bounds 2`] = `
<yoo-form-slider
  class="hydrated"
  data-yoo-form-slider-host=""
  initial-value="34"
>
  
  <div
    class="wrapper"
    data-yoo-form-slider=""
  >
    
    <div
      attr-layout="flex"
      class="outer-container"
      data-yoo-form-slider=""
    >
      <label
        data-yoo-form-slider=""
      >
        0
      </label>
      <div
        attr-layout="column"
        class="slider-container"
        data-yoo-form-slider=""
      >
        <div
          class="label-value"
          data-yoo-form-slider=""
        >
          <input
            data-yoo-form-slider=""
            type="number"
          />
        </div>
        <div
          attr-layout="row"
          class="range-container"
          data-yoo-form-slider=""
        >
          <input
            class="swiper-no-swiping"
            data-yoo-form-slider=""
            max="100"
            min="0"
            type="range"
          />
          
          <yoo-progress-bar
            class="hydrated"
            data-yoo-form-slider=""
            hide-progress="true"
            progress="0"
          />
        </div>
        
      </div>
      <label
        data-yoo-form-slider=""
      >
        100
      </label>
    </div>
  </div>
</yoo-form-slider>
`;
